### PR TITLE
Raise minimum Ruby version to 2.7

### PR DESCRIPTION
**Pull request by archanaserver**
This contain changes of ruby version support, for `>= 2.7', '< 4`

### CODE REVIEW

**Comment by archanaserver** [comment]
@ekohl just raised this pr to update ruby version, this was way outdated https://github.com/theforeman/ldap_fluff/pull/75

File: Gemfile [hunk 1]
```diff
@@ -2,6 +2,6 @@
 
 source 'https://rubygems.org'
 
-gemspec
+gemspec :name => 'ldap_fluff'
```
Comment (ekohl) [inline Gemfile:5]: Why is this needed?

File: Gemfile [hunk 2]
```diff
@@ -2,6 +2,6 @@
...
 
-gem 'theforeman-rubocop', '~> 0.0.6', :group => :test
+gem 'theforeman-rubocop', github: 'archanaserver/theforeman-rubocop', branch: 'rubocop'
```
Comment (ekohl) [inline Gemfile:7]: This shouldn't be in this PR.

**Review by ekohl — CHANGES_REQUESTED** [review]
The summary of this commit should be `Raise minimum Ruby version to 2.7`. Typically those kinds of changes are major version bumps.

**Comment by archanaserver** [comment]
@ekohl ah i marked it as wip and so pushed random change to check the gha workflow result for the rubocop, because rake don't have rule to check rubocop result. do you want me to work on that first?

**Comment by ekohl** [comment]
If it's WIP then using a draft PR makes it clear to reviewers.

**Comment by ekohl** [comment]
> because rake don't have rule to check rubocop result. do you want me to work on that first?

You can also override the command provided to https://github.com/theforeman/actions/blob/v0/.github/workflows/rubocop.yml. We can also change the action. I recall @evgeni mentioning something similar because you can't enable parallel testing from Rakefile.

The reason I preferred the Rakefile approach in the action was that we very often had old versions of RuboCop that either didn't have the command line utility at all (that came later) or was too old to support `--parallel`.

By now I think we can start to enforce use of recent enough versions of RuboCop instead of keeping compatibility.

**Comment by archanaserver** [comment]
This needs rebase once, #75 gets merged!

**Comment by ekohl** [comment]
https://github.com/theforeman/ldap_fluff/pull/75 includes this and passed, so merging.

**Comment by archanaserver** [comment]
> #75 includes this and passed, so merging.

Thank you @ekohl

**Pull request merged**